ci: tags: Make sure openthread tag is used on OT changes

rlubos opened the pull request:
Use namespaced variants instead.

Note: I have purposedly skipped one symbol to see if CI triggers and catches the issue this time. Will update after getting results.

rlubos committed:
MAINTAINERS: Specify OpenThread tests for OpenThread module

Make sure that OpenThread tests are listed in OpenThread module
area.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

aescolar commented:
> Note: I have purposedly skipped one symbol to see if CI triggers and catches the issue this time

@rlubos it did not :(

rlubos committed:
ci: tags: Add folders covered by openthread tag

openthread tag was used in test specifications, but did not define what
source directories were related to this tag.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
modules: openthread: Breaking change

Breaking change in OpenThread module to trigger CI failure.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
What I'm trying to achieve here is that changes in `modules/openthread` trigger twister tests with `openthread` tag so that we prevent regressions like earlier today.

I'm kind of out of ideas here, where should I specify the directories so that CI runs the twister with `openthread` tag? What I did so far doesn't seem to work with this PR. @nashif Any help here?

nashif commented:
> What I'm trying to achieve here is that changes in `modules/openthread` trigger twister tests with `openthread` tag so that we prevent regressions like earlier today.
> 
> I'm kind of out of ideas here, where should I specify the directories so that CI runs the twister with `openthread` tag? What I did so far doesn't seem to work with this PR. @nashif Any help here?

yeah, looking at that. Tags right now only get excluded, not included. This was limited to excluding after we had some issues with coverage at some point.
